### PR TITLE
feat(sections): beat-precise section boundaries

### PR DESCRIPTION
**Pull request by forgerpl**
Sections were locked to measure lines, but real songs change parts mid-measure — a chorus
that starts on beat 3, an outro that begins half a beat early.

- `start_beat` / `end_beat` (1-based within the boundary measure, fractional, omitted
  meaning the measure line) offset either end to any position on the beat grid.
- `BeatGrid::beat_time` resolves a fractional position by interpolating between grid
  beats; `resolve_section` uses it, so seek-to-section and section loops honour the
  offsets with no further change.
- Validation orders full (measure, beat) positions, so a section may now live inside a
  single measure when its beats are ordered.
- Three ways to set one without counting beats: the dialog's start/end beat steppers, an
  edge drag that snaps to beats once the zoom makes them distinct (measure lines when
  zoomed out), and — when the song is loaded in the player — capturing the preview
  playhead with **Start here** / **End here**, which refuses a capture that would invert
  the section.

Docs and two captures in the last commit: a section starting on beat 3 and ending on beat
2½, and the dialog with its beat steppers and the capture row.

section-beat-boundaries

section-dialog

### CODE REVIEW

File: src/config/song.rs [hunk 1]
```diff
@@ -210,10 +218,29 @@ impl Song {
...
+                        ));
+                    }
+                }
+            }
+            // The end bound must come after the start; equal measures are
+            // fine when the beats are ordered (a section within one measure).
+            let start_beat = section.start_beat.unwrap_or(1.0);
+            let end_beat = section.end_beat.unwrap_or(1.0);
+            if section.end_measure < section.start_measure
+                || (section.end_measure == section.start_measure && end_beat <= start_beat)
```
Comment (mdwn) [inline src/config/song.rs:261]: This ordering test compares positions as `(measure, beat)` tuples, which is only sound while a beat stays inside its measure — and nothing enforces that.

`beat_time` indexes the flat beat array (`measure_starts[measure] + floor(beat - 1)`), so a beat past the measure's length silently spills into the next measure. Validation can't see the grid, so it has no idea how many beats a measure holds, and the dialog's steppers offer `min={1} max={32}` with no awareness of the meter.

Concretely, in 4/4 (grid beats every 0.5s, `measure_starts [0, 4]`), a section with `start_measure: 1, start_beat: 5, end_measure: 2`:

- start = `beat_time(0, 5.0)` = `beats[4]` = **2.0s**
- end = `beat_time(1, 1.0)` = `beats[4]` = **2.0s**

This check passes (`end_measure 2 > start_measure 1`) and the section is zero-length; `start_beat: 6` inverts it outright. Downstream `SectionLoopTrigger::check` does `end.saturating_sub(start)` → zero → returns `None`, so **the section loop silently never fires** — no error, no log, arming just does nothing, while seek-to-section still jumps somewhere and makes it look half-working.

"Beat 5 in 4/4" isn't exotic — it's the natural way to say "the downbeat of the next measure", and the stepper presents it as valid up to 32.

Worth noting the drag path already gets this right: `SectionBar` compares *resolved times* (`candidate < posToMs(section.end_measure, section.end_beat)`), and `snapBoundary` derives positions from the grid so its beats never exceed their measure. It's the dialog and this validator that are exposed — and the dialog's `isBefore` (used to gate the **Start here** / **End here** capture buttons) shares the same blind spot, so that inversion guard is defeatable too.

Any of these would close it:

- Clamp the beat steppers' `max` to the measure's beat count — `SectionTimelineEditor` has `song.beat_grid` and already computes measure spans.
- Compare resolved times rather than tuples in the dialog's `isBefore`, matching what the drag path does.
- Defensively, have `resolve_section` return `None` when `start >= end`; config validation can't see the grid, but `resolve_section` can.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by mdwn — COMMENTED** [review]
Nice change — `beat_time` is a clean primitive, and threading it through `resolve_section` with `unwrap_or(1.0)` means seek and section loops pick up the offsets with genuinely no other changes and no behavior drift when beats are absent.

One issue inline: a start beat past the measure's own length spills into the next measure, and because validation (and the dialog's capture guard) order positions as `(measure, beat)` tuples rather than resolved times, that can produce a zero-length or inverted section which passes validation and then silently refuses to loop. The drag path already compares resolved times correctly, so it's mostly a matter of making the other two agree with it.

Everything else checks out: `beat_time` handles the fractional/boundary cases correctly (final grid beat stays addressable, out-of-grid returns `None`, non-finite rejected), the within-one-measure relaxation is deliberate and tested, and `cargo test --lib section` plus `beat_grid_beat_time` are green (58 + 1 passing).

Also, for anyone reading the file list: the `mcp/` and `CHANGELOG` deletions shown in the diff aren't a revert — the branch forked at `3954132` and main has since merged `6f0401f`. Against the true merge-base the diff is clean.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/config/song.rs (hunk 1 above)
Comment (forgerpl) [reply src/config/song.rs:261]: Good catch, and it went further than the zero-length case - thanks, fixed.

**Review by mdwn — APPROVED** [review]
